src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
```

Comment:
Oh, I understand the issue now. Maybe we could use something like [managed](https://github.com/m-labs/rust-managed) to allow users to opt-in to using references? I'm not sure how difficult that would be.

Anyways, that doesn't need to be solved in this PR. I'll open an issue so we can track these ideas